elastic/kibana
# [Snapshot Restore] Fix broken Snapshot restore form

Fix https://github.com/elastic/kibana/issues/160974
Fix https://github.com/elastic/kibana/issues/160929

## Summary

From Es 8.10, the `version` field from the Get Snapshot response has a different format - it will no longer be a string representing the release version, but rather it will be a string representing the version id (see https://github.com/elastic/kibana/issues/160974#issuecomment-1618087194 for more details).

This PR replaces the use of the `version` field with the `versionId` field when checking in the Snapshot Restore form if the current version is greater than 7.12 (which corresponds to `versionID` 7120099).
The PR also unskips the functional test that failed with the promotion of Es to 8.10.

**How to test:**

1. Start Es with `yarn es snapshot -E path.repo=./tmp/snap` and Kibana with `yarn start`
2. Go to Stack Management -> Snapshot and Restore
3. Create a Shared File System repository with location `./tmp/snap`
4. Create a policy that uses the repository created in the previous step.
5. Run the policy to create a snapshot.
6. Go to the Snapshots tab and click the Restore button for the created snapshot.
7. Verify that the Snapshot restore form works as expected.
